Feature/ec2 vpc subnet

wimnat opened the pull request:
Module by @erydo rebased to latest ansible-modules-extras

For review by @defionscode 

Obsoletes PR https://github.com/ansible/ansible-modules-core/pull/322

willthames commented on cloud/amazon/ec2_vpc_subnet.py:
```diff
@@ -0,0 +1,249 @@
+        module.fail_json(msg='Region must be specified')
+
+    try:
+        vpc_conn = boto.vpc.connect_to_region(
```
Use get_aws_connection_info in conjunction with connect_to_aws(boto.vpc,...)

juliedavila commented on cloud/amazon/ec2_vpc_subnet.py:
```diff
@@ -0,0 +1,249 @@
+    default: null
+  resource_tags:
+    description:
+      - A dictionary array of resource tags of the form: { tag1: value1, tag2: value2 }. This module identifies a subnet by CIDR and will update the subnet's tags to match. Tags not in this list will be ignored.
```
This should be quoted or ansible-doc cannot parse it b/c of the curlies

```
    description:
      - "A dictionary array of resource tags of the form: { tag1: value1,
        tag2: value2 }. This module identifies a subnet by CIDR and will update
the subnet's tags to match. Tags not in this list will be ignored."
```

juliedavila commented:
`region` should be added to the doc string as it is required with "normal" usage.

Overall, this looks great, but a couple of style suggestions.

Currently the returned value of creating a subnet is:

```
changed: [localhost] => {"changed": true, "subnet": {"az": "us-east-1b", "cidr": "10.0.1.0/28", "id": "subnet-0aea7121", "tags": {"Name": "Frank", "delete": "me"}}, "subnet_id": "subnet-0aea7121"}
```

As you can see the keys `id`and `subnet_id` are redundant. This should probably be patched so it only returns one or the other (my preference is for `subnet_id`).  

Lastly, I prefer to see a standalone `name` parameter. This way the end user doesn't have to remember to create a `Name: $value` under the `resource_tags` parameter. Like this:

``` yml
- name: Create subnet for database servers
  ec2_vpc_subnet:
    state: present
    name: database_subnet
    vpc_id: vpc-123456
    cidr: 10.0.1.16/28
    resource_tags:
      Name: Database Subnet
  register: database_subnet
```

Not a huge thing, would not hold off on merging from this alone, just a style thing.

halberom commented:
I'm getting `changed: True` for repeated runs. Actual json output appears to be identical.

wimnat commented:
@halberom can you share your playbook please?

halberom commented:
https://github.com/HighOps/ansible_ec2_vpc_nat_asg

wimnat commented:
@halberom i think i got to the bottom of your changed=true problem. This module was maintaining existing tags when comparing to what was defined under 'tags' in the module. The module would maintain the old tags so the comparison would always be unequal and force a change.

I've removed this - now only the tags defined here are applied.  Tags that exist already but not defined in the module will be removed. I know some people may not like this destructive nature but Ansible is about 'Desired State Configuration' not a script to add some tags and preserve what already exists.

@defionscode I've added more detail to what is returned and eliminated the duplicate ID, replicating what exists in my facts module https://github.com/ansible/ansible-modules-extras/pull/783.  If both these modules pass muster then i'd appreciate the +1 to finally get this merged!

juliedavila commented:
+1
